platform/socket.hpp

```diff
@@ -26,5 +26,21 @@ class Socket
   virtual void SetTimeout(uint32_t milliseconds) = 0;
 };
 
+class TestSocket : public Socket
```

Comment:
Please, move TestSocket to platform_tests_support. Or, even better, to the platform tests directory.